ledger/src/blockstore_processor.rs

```diff
@@ -921,21 +920,57 @@ pub fn confirm_slot(
 
     let verify_transaction = {
         let bank = bank.clone();
-        move |versioned_tx: VersionedTransaction| -> Result<SanitizedTransaction> {
-            bank.verify_transaction(versioned_tx, skip_verification)
+        move |versioned_tx: VersionedTransaction,
+              verification_mode: entry::EntryVerificationMode|
+              -> Result<SanitizedTransaction> {
+            let result = bank.verify_transaction(
+                versioned_tx,
+                verification_mode == entry::EntryVerificationMode::HashOnly
+                    || verification_mode == entry::EntryVerificationMode::HashAndVerifyPrecompiles,
```

Comment:
Could you use an enum or options struct for `Bank::verify_transaction` as well?

Reply:
As Bank::verify_transaction is used in other places, and right now, this is the only code that requires the ability to verify precompiles separately from verifying the signatures, I would prefer not to change Bank::verify_transaction at this time, to keep this change focused, and to avoid adding extra complexity to Bank::verify_transaction when it would only be used in one place. If nothing else, in my opinion, such as refactoring would be the scope of a separate change.

Reply:
@jstarry thanks for the suggestion - have implemented it by adding an using an enum for the transaction verification.